components/scream/src/physics/p3/p3_functions_cloud_rain_acc_impl.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef P3_FUNCTIONS_CLOUD_RAIN_ACC_IMPL_HPP
+#define P3_FUNCTIONS_CLOUD_RAIN_ACC_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace p3 {
+
+/*
+ * Implementation of p3 cloud rain accretion function. Clients should NOT
+ * #include this file, but include p3_functions.hpp instead.
+ */
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::cloud_rain_accretion(const Spack& rho, const Spack& inv_rho,
+                       const Spack& qc_incld, const Spack& nc_incld,
+                       const Spack& qr_incld, Spack& qcacc, Spack& ncacc)
+{
+  constexpr Scalar qsmall = C::QSMALL;
+
+  const auto qr_and_qc_not_small = (qr_incld >= qsmall) && (qc_incld >= qsmall);
+  if (qr_and_qc_not_small.any()) {
+    constexpr int iparam = P3C::iparam;
+
+    if (iparam == 1) {
```

Comment:
We have decided to only use option iparam=3, so the other options don't need to be included.